container/build.py

```diff
@@ -115,32 +119,54 @@
 def main(type_, project, container, tag, async_, bucket, local, bazel_version, map = None):
   '''Runs the build. More info in module docstring at the top.
   '''
-  
+  type_package_map = TYPE_PACKAGE_MAP
+  type_target_map = TYPE_TARGET_MAP
+  type_tarball_map = TYPE_TARBALL_MAP
+
   if map: # Override the map values
-    map_module = imp.load_source("map", map)
-    TYPE_PACKAGE_MAP = map_module.TYPE_PACKAGE_MAP
-    TYPE_TARGET_MAP = map_module.TYPE_TARGET_MAP
-    TYPE_TARBALL_MAP = map_module.TYPE_TARBALL_MAP
+    try:
+      map_module = imp.load_source("map", map)
+    except IOError as e:
+      print("Error reading map file.\n" ,e)
+    
+    try:
+      type_package_map = map_module.TYPE_PACKAGE_MAP
+      type_target_map = map_module.TYPE_TARGET_MAP
+      type_tarball_map = map_module.TYPE_TARBALL_MAP
+    except AttributeError as e:
+      print("Error getting attributes from map file.\n", e)
 
   # Gets the project root (for calling bazel targets)
   project_root = subprocess.check_output(
       shlex.split("git rev-parse --show-toplevel")).strip()
-  package = TYPE_PACKAGE_MAP[type_]
-  target = TYPE_TARGET_MAP[type_]
+  package = type_package_map[type_]
+  target = type_target_map[type_]
   tarball = None
   if bucket:
-    tarball = TYPE_TARBALL_MAP[type_]
-
-  # Gets the base directory of the bazel-toolchains repo relative to this 
-  # build.py. This is for referencing yaml files and mounting project to gcloud
-  # THIS NEEDS TO BE UPDATED IF THIS FILE IS MOVED 
-  bazel_toolchains_base_dir = os.path.relpath(os.path.join(__file__, "../.."))
-
+    tarball = type_tarball_map[type_]
+
+  # Gets the base directory of the bazel-toolchains repo (relative to project_root)
+  # This is for referencing yaml files and mounting the project to gcloud.
+  # Allows the bazel-toolchains repo to be cloned in any subdirectory
+  # of another project and bazel_toolchains_base_dir will store the relative path
+  # from the root of that git project to the root of the bazel-toolchains git project.
+  # Ex. If we are in folder foo and the structure looks like this:
+  # foo/
+  #     subdir/
+  #         bazel-toolchains
+  #     ...
+  # 
+  # Then if we call build.py with the terminal running in foo,
+  # bazel_toolchains_base_dir == "subdir/bazel-toolchains"
+  # (This also allows for renaming of the bazel-toolchains folder as the variable
+  # will store the path regardless of the directory names) 
+  os.chdir(os.path.dirname(__file__))
+  bazel_toolchains_base_dir = os.path.relpath(subprocess.check_output(
```

Comment:
just to confirm the git root obtained here is the git_root of bazel-toolchains repo right?